api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -333,6 +333,15 @@ default UpdateStatistics updateStatistics() {
    */
   Map<String, SnapshotRef> refs();
 
+  /**
+   * Returns the UUID of the table
+   *
+   * @return the UUID of the table
+   */
+  default String uuid() {
+    throw new UnsupportedOperationException(this.getClass().getName() + " doesn't implement uuid");
```

Comment:
This was done to avoid breaking API changes. I don't expect users to override `Table` due to the presence of core implementations and if they do, and want to use `UUID` they should implement that.